virt_who/provision.py

```diff
@@ -2057,9 +2057,17 @@ def vcenter_guest_uuid(self, cert, ssh_vcenter, guest_name):
         cmd = "%s Get-VM %s | %%{(Get-View $_.Id).config}" % (cert, guest_name)
         ret, output = self.runcmd(cmd, ssh_vcenter)
         if ret == 0:
+            for line in output.splitlines():
+                if re.match(r"^Version.*:", line):
+                    version = line.split(':')[1].strip()
+                    logger.info("Succeeded to get the VMware version: {0}".format(version))
             for line in output.splitlines():
                 if re.match(r"^Uuid.*:", line):
-                    uuid = line.split(':')[1].strip()
+                    s = line.split(':')[1].strip()
+                    if version > "vmx-13":
+                        uuid = s[6:8] + s[4:6] + s[2:4] + s[0:2] + "-" + s[11:13] + s[9:11] + "-" + s[16:18] + s[14:16] + "-" + s[19:]
+                    else:
+                        uuid = s
```

Comment:
is it possible to combine these two `for` loop to one? such as:
```
            for line in output.splitlines():
                if re.match(r"^Version.*:", line):
                    version = line.split(':')[1].strip()
                if re.match(r"^Uuid.*:", line):
                    s = line.split(':')[1].strip()
            if version > "vmx-13":
                uuid = s[6:8] + s[4:6] + s[2:4] + s[0:2] + "-" + s[11:13] + s[9:11] + "-" + s[16:18] + s[14:16] + "-" + s[19:]
            else:
                uuid = s
            return uuid
```

Reply:
Thanks for your suggestion to simplify the code and I have updated it.